packages/nextjs/src/middleware/locale-middleware.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
+
+import { NextResponse, NextRequest } from 'next/server';
+import { debug } from '@sitecore-content-sdk/core';
+import { getLocaleRewrite } from '@sitecore-content-sdk/core/i18n';
+import { MiddlewareBase, MiddlewareBaseConfig, LOCALE_HEADER_NAME } from './middleware';
+
+export type LocaleMiddlewareConfig = MiddlewareBaseConfig & {
+  locales: string[];
+};
+
+export class LocaleMiddleware extends MiddlewareBase {
+  private locales: string[];
+  /**
+   * @param {LocaleMiddlewareConfig} [config] Locale middleware config
+   */
+  constructor(protected config: LocaleMiddlewareConfig) {
+    super(config);
+    this.locales = config.locales;
```

Comment:
It's not neccessary to set "locales" we can just read them directly from "config"